[Java] Use latest AE in the server and ref-app (#221)
godaddy/asherah: samples/java/reference-app/README.md

Example run using MySQL metastore, AWS KMS, CloudWatch metrics and 100 iterations:

```console
[user@machine reference-app]$ java -jar target/referenceapp-1.0.0-SNAPSHOT-jar-with-dependencies.jar \
--metastore-type JDBC \
--jdbc-url 'jdbc:mysql://localhost/test?user=root&password=password' \
--kms-type AWS \

>  --preferred-region us-west-2 \

Reviewer: ```suggestion
--preferred-region us-west-2 \
```

 --provisioned-throughput \
   ReadCapacityUnits=1,WriteCapacityUnits=1
```

#### Global Tables

To use Global Tables, the above table needs to be created with few modifications.

> More details about how to create a Global Table can be found [here](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/globaltables.tutorial.html)

Reviewer: ```suggestion
More details about how to create a Global Table can be found in the [AWS Developer Guide](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/globaltables.tutorial.html)
```

Example run using DynamoDB metastore with key suffixes enabled (for Global Tables), a custom table name, and a
local DynamoDB endpoint.

```console
[user@machine reference-app]$ java -jar target/referenceapp-1.0.0-SNAPSHOT-jar-with-dependencies.jar \
--metastore-type DYNAMODB \
--enable-key-suffix us-west-2 \
--dynamodb-endpoint http://localhost:8000 \
--dynamodb-signing-region us-west-2 \
--dynamodb-table-name MyGlobalTable
``` 

## Configuring the Reference App
Configuration options are provided via command-line arguments. Supported options are as
follows:

```console
--drr=<drr>          DRR to be decrypted
--dynamodb-endpoint=<endpoint>
                     The DynamoDb service endpoint (only supported by DYNAMODB)
--dynamodb-region=<region>
                     The AWS region for DynamoDB requests (only supported by DYNAMODB)
--dynamodb-signing-region=<signingRegion>
                     The DynamoDb service endpoint (only supported by DYNAMODB)
--dynamodb-table-name=<dynamoDbTableName>
                     The table name for DynamoDb (only supported by DYNAMODB)

> --enable-cw          
>                      Enable CloudWatch Metrics output

Reviewer 2: ```suggestion
--enable-cw          Enable CloudWatch Metrics output
```
